beacon-chain/rpc/beacon/validators.go

```diff
@@ -802,15 +802,25 @@ func (bs *Server) GetValidatorPerformance(
 	correctlyVotedHead := make([]bool, 0, reqPubKeysCount)
 	missingValidators := make([][]byte, 0, reqPubKeysCount)
 
-	headState, err := bs.HeadFetcher.HeadState(ctx)
-	if err != nil {
-		return nil, status.Error(codes.Internal, "Could not get head state")
-	}
+    headState, err := bs.HeadFetcher.HeadState(ctx)
+    if err != nil {    
+        return nil, status.Errorf(codes.Internal, "could not get head state: %v", err)
+    }
+ 
 	// Convert the list of validator public keys to list of validator indices.
 	// Also track missing validators using public keys.
 	for _, key := range req.PublicKeys {
 		pubkeyBytes := bytesutil.ToBytes48(key)
 		idx, ok := headState.ValidatorIndexByPubkey(pubkeyBytes)
+		val, err := headState.ValidatorAtIndex(idx)
+		currentEpoch := helpers.CurrentEpoch(headState)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not get validator: %v", err)
```

Comment:
fix the error here 
```
could not get current epoch
```

Reply:
Oh, why that is happening I saw this code working in other places?
Is the way I am getting it wrong?

Reply:
ok so your order of code here is vulnerable to errors here. if you look at 
```go
		idx, ok := headState.ValidatorIndexByPubkey(pubkeyBytes)
```
you need to handle the case where the index doesn't exist. 
so immediately after that you need to do:
```go
		if !ok {
			missingValidators = append(missingValidators, key)
			continue
		}
```
Then  doing this: 
```go
		val, err := headState.ValidatorAtIndex(idx)
		currentEpoch := helpers.CurrentEpoch(headState)
		if err != nil {
```
is wrong you need to handle the error first 
```go
		val, err := headState.ValidatorAtIndex(idx)
		if err != nil {
                        return nil, status.Errorf(codes.Internal, "could not get validator: %v", err)
                }		
               currentEpoch := helpers.CurrentEpoch(headState)
```